flycheck.el

```diff
@@ -5596,6 +5597,30 @@ See URL `http://phpmd.org/' for more information about phpmd."
                       errors)))))))))
        (nreverse errors)))))
 
+(defun flycheck-parse-reek (output checker buffer)
+  "Parse Reek warnings from JSON OUTPUT.
+
+CHECKER and BUFFER denoted the CHECKER that returned OUTPUT and
+the BUFFER that was checked respectively.
+
+See URL `https://github.com/troessner/reek' for more information
+about Reek."
+  (let ((errors))
+    (dolist (message (car (flycheck-parse-json output)))
+      (let-alist message
+        (dolist (line (delete-dups .lines))
```

Comment:
Manual :)
https://www.gnu.org/software/emacs/manual/html_node/elisp/Sets-And-Lists.html#Sets-And-Lists

Sometimes you can have same error twice on same line. Then you get output like this:
```javascript
[{"context":"ApplicationController#validate_type","lines":[15,15,16],"message":"calls 'params['data']' 3 times","smell_type":"DuplicateMethodCall","source":"app/controllers/application_controller.rb","name":"params['data']","count":3,"wiki_link":"https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md"}]
```
Line 15 is two times there.